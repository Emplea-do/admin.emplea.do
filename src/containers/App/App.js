import React, { useState } from 'react'

import Button from '../../component/forms/Button'

function App() {
  const [count, setCount] = useState(0)
  return (
    <>
      <h1>
        Clicks: <span data-testid="counter">{count}</span>
      </h1>
      <Button onClick={() => setCount(count => count + 1)}>Add</Button>
    </>
  )
}

export default App
