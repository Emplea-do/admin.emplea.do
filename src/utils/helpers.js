export const tap = val => console.log(val) || val
