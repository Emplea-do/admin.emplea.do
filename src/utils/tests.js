import React from 'react'
import { ThemeProvider } from 'emotion-theming'
import { Global } from '@emotion/core'
import { render } from '@testing-library/react'

import { globalCSS, theme } from '../styles'

export const renderWithApp = (ui, options) =>
  render(
    <>
      <Global styles={globalCSS} />
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </>
  )
