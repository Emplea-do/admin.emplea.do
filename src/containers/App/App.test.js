import React from 'react'
import { fireEvent } from '@testing-library/react'

import { renderWithApp } from '../../utils/tests'

import App from './App'

describe('containers/App', () => {
  test('works correctly', () => {
    const { getByText, getByTestId } = renderWithApp(<App />)
    const button = getByText(/add/gi)
    const countLabel = getByTestId(/counter/gi)
    expect(countLabel).toHaveTextContent('0')
    fireEvent.click(button)
    fireEvent.click(button)
    fireEvent.click(button)
    expect(countLabel).toHaveTextContent('3')
  })
})
