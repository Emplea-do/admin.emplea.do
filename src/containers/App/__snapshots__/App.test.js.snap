// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/App works correctly 1`] = `[Function]`;
